### ISSUE TITLE
PHP 8.0

### ISSUE THREAD

**Issue by tonodor** [iuscommunity/wishlist#294]
**What new package do you want?**

PHP 8.0

**Why?**

Support for the latest PHP version on RHEL / CentOS 7

**Testing**

I agree to test the new package to ensure that it works as expected.  Once I am satisfied with the results of my testing I will comment on this issue with the word "STABLE" to get it promoted to the stable repos.

**Comment by rossnick** [comment]
Is there any news on this ? Since PHP as a more rapid cycle of developpement, php 7.4 is only in active support until the end of the year, this is needed...

**Comment by lubo** [comment]
Fedora 35 has been released, so there is no other blocker, right?

**Comment by rossnick** [comment]
@lubo with the phase-out of centos 7 and the rise of Rocky Linux 8 on our production server, and since RL8 now uses appstreams instead of repo, I unfortunately gave up on IUS in favor of remi's modular repo that have php 8.0 and even 8.1 availaible.

**Comment by srguglielmo** [comment]
Yes, I believe [RemiRepo](https://rpms.remirepo.net/) is now the best choice for PHP > 7.4. on RHEL/CentOS.

Much appreciation to IUS for providing packages through the years!

**Comment by lubo** [comment]
Actually, Remi's repository seems to be largely a one-man show and the handling of remicollet/remirepo#158 is IMO unsatisfactory, which means it's probably not the best choice from the business continuity standpoint.

@rossnick If you can easily upgrade from CentOS 7, then perhaps you can easily upgrade to CentOS Stream 9, which features PHP 8.0 out of the box. Or perhaps you can wait for RHEL 9, which will be released in early 2022 I think. Anyway, IUS does not intend on supporting RHEL 8 or newer, so there is no point in discussing that here.

@jeffsheltren My organisation is potentially interested in maintaining this, provided that we can use your infrastructure. What would be the next step?

**Comment by rossnick** [comment]
All of our new installation since last summer uses Rock Linux 8, but they are "infra" server mostly using docker. So our web app are now deployed using a custom-built docker image base on Rock with remirepo appstream php 7.4 and 8. So switching to RHEL9 or other php repo should be relativly easy.

**Comment by carlwgeorge** [comment]
FWIW, both CentOS Stream 8 and CentOS Stream 9 now have PHP 8.0 included in the main distribution, no third party repositories necessary.

I don't know if IUS will ever provide a php80 package.  php74 took significant effort and coordination with EPEL to do incompatible version upgrades of some libraries as well as ship new libraries.  Just identifying what all would be needed for php80 would be a significant task.  Shipping the latest version of any software gets increasingly difficult as the base operating system ages.  EL7 is now 7 years and 7 months old.  It's based on a Fedora release that came out 8 years and 6 months ago.

I'm not blocking anyone from contributing a php80 package to IUS, I just haven't had the time to do it myself.  I have written a [packaging walkthrough](https://github.com/iusrepo/packaging/wiki/Walkthrough) to help new contributors create packages that follow the [IUS packaging guidelines](https://github.com/iusrepo/packaging/wiki/Guidelines).  Personally I think that that effort would be better spent migrating to C8S or C9S, but I'm happy to provide some guidance if anyone just has to make this work on EL7.

**Comment by rossnick** [comment]
Thanks @carlwgeorge for the follow-up. CentOS Steam is a no-go for me. Since the end of 2021, like I mentionned we are using Rocky Linux as a base image for our docker containers that serves PHP (and for a few web server directly also), using Remi's repos, even though I initially didn't like Remi's packagings of PHP when I initially discovered and used IUS, probably for not that good reasons, but maybe some good.

Since then, php 8.1 has been released and now have made availaible to our devs container for php 8.1 in a matter of days, since 8.1 was added to Remi's repo relativly quickly.

I don't like that the Remi's repo is probably a one-person adventure, what will happen if he looses interest in maintaining the repo ??

While I do appreciate the usefullness and the effort that has been put into IUS, and we will still use it for our current CentOS 7 servers until they are replaced with docker images or Rocky 8 (with Remi's repo), I don't see our company using IUS for new installation from now on.
